modules/fedimint-gw-client/src/receive.rs

```diff
@@ -0,0 +1,15 @@
+/// State machine that handles an intercepted HTLC from the Lightning node.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    HTLCIntercepted -- await preimage offer accepted --> BuyPreimage
+///    HTLCIntercepted -- await preimage offer failed --> CancelHTLC
```

Comment:
This failure means "offer not found"?

Reply:
Added a new state/transition to make this more clear